### DISCUSSION TITLE
Crash by selecting non-existent text

### DISCUSSION THREAD

**Discussion by soopax** [ppy/osu#23614] [Q&A]
# Steps to reproduce
1. Import a beatmap and open it in the editor.
2. Go to the setup tab.
3. Select some text with the cursor. Don't make any changes to the beatmap. (See this screenshot for an example:)
![Screenshot from 2023-05-21 15-35-15](https://github.com/ppy/osu/assets/101510334/ad288361-1392-41c6-b465-2538a4c37522)
4. Don't move the cursor. Keep the left mouse button pressed on the text selection.
5. Press `Esc` twice to exit the editor. Your cursor should still be pink.
6. Move the cursor again.

This also happens with sliders or spinners in the compose tab.
1. Select a hit circle or slider
2. Don't move the cursor. Keep the left mouse button pressed on the object.
3. Press `Esc` to exit the editor. Your cursor should still be pink.
4. Move the cursor again.

Log:
[runtime.log](https://github.com/ppy/osu/files/11524574/runtime.log)

Edit: clarification

Edit: compose tab log:
[runtime.log](https://github.com/ppy/osu/files/11525112/runtime.log)

**Comment by bdach** [comment]
https://github.com/ppy/osu/issues/23616 for the textbox case. logs are missing for the composer case

**Reply by soopax** [reply]
Edited to add the log for the compose tab case.
